batch/batch/driver/scheduler.py

```diff
@@ -156,8 +157,21 @@ def allocate_cores(user, mark):
                 (user_record['user'],))
             async for record in records:
                 should_wait = False
-                await unschedule_job(self.app, record)
 
+                async def unschedule_with_error_handling(id, instance, f):
+                    try:
+                        await f()
+                    except Exception:
+                        log.info(f'unscheduling job {id} on instance {instance}', exc_info=True)
+                async_work.append(
+                    functools.partial(
+                        unschedule_with_error_handling,
+                        id, record['instance_name'],
+                        functools.partial(
+                            unschedule_job,
+                            self.app, record)))
```

Comment:
I generally find it easier to read if we bake in the `unschedule_job` above and pass its arguments to the `unschedule_with_error_handling`, but this is fine too.